app/contexts/Auth.tsx

```diff
@@ -21,7 +22,14 @@ export default function AuthProvider({
       clientId={CLIENT_ID}
       authorizationParams={{
         redirect_uri: redirect_uri || "",
-        scope: "openid profile email",
+        /**
+         * Note that offline_access is requried for refresh tokens -
+         * the alternative is to use a custom hook that occasionally
+         * calls getAccessTokenSilently(), which will manually refresh
+         * the token
+         */
+        scope: "openid profile email offline_access",
+        audience: AUDIENCE,
```

Comment:
while testing this i realized that our tokens were not refreshing—this fixes that. i spent some time staring at the old VZE and couldn't quite figure out how we are refreshing tokens in that app.

Reply:
our tokens last for 10 hours, so before adding this, id have to sign in every 10 hours?
or the query wouldnt work but refreshing the app would fix it, because it would refresh the token without offline_access?

Reply:
i believe our tokens expire after 24 hours. before this change, your token would expire and all of your API calls would fail, but you would still be logged into the app. if you signed into the netlify deployment more than 24 hours ago you can see this for yourself: https://vision-zero-nextjs.netlify.app/crashes

to resolve it, you have to manually sign out and sign back in to the app.

so, yeah, as long as you have a session token in local storage, the Auth0 library just assumes it's valid and the `useAuth0` hook returns a normal-looking `user` object with `isAuthenticated = true`.

If we wanted to fix this without using refresh tokens, we could  use a `try...catch` block with  `getAccessTokenSilently()`, which would fail and then we could redirect users to the login screen.

With refresh tokens, the app is going to grab a new token for you whenever you refresh the page. I'm still not 100% if your token expires before you refresh it—we may still need to do the `try...catch` thing to make sure users get sent back to login.

Reply:
and more thing—the reason this isn't obvious on local dev is because we don't run the local stack in JWT mode. the endpoints are not pw protected at all. we might want to take this opportunity to change that.

Reply:
and answering part of your question that i missed: the `offline_access` scope is a special Auth0 security permission that must be togged on in order to allow refresh tokens—i did not bother to understand why. it's not just adding it here to the `scope` that was needed, i had to flip a toggle for it in our Auth0 app settings.

Reply:
oh I dont know if this is has been mentioned elsewhere, but I did get that alert from auth0 that they were requesting the new offline_access scope, could be nice to alert our users when the nextjs upgrade gets rolled out to production, to expect to grant new access. because they will see the same alert, right?

Reply:
Thanks for that reminder. Yes and I created https://github.com/cityofaustin/atd-data-tech/issues/20311 to remember things like this 🙏